src/wallet/wallet.h

```diff
@@ -1178,8 +1178,8 @@ class CWallet final : public WalletStorage, private interfaces::Chain::Notificat
     ScriptPubKeyMan* GetScriptPubKeyMan(const uint256& id) const;
 
     //! Get the SigningProvider for a script
-    std::unique_ptr<SigningProvider> GetSigningProvider(const CScript& script) const;
-    std::unique_ptr<SigningProvider> GetSigningProvider(const CScript& script, SignatureData& sigdata) const;
+    std::unique_ptr<SigningProvider> GetSolvingProvider(const CScript& script) const;
```

Comment:
A little confused by this commit. Should `SigningProvider` itself be renamed as well?

Reply:
We should probably refactor `SigningProvider` at some point to have a parent `SolvingProvider` that doesn't have private keys. But for now, this just returns a `SigningProvider` without private keys.

Reply:
I have commit for exactly that, see my earlier comment: https://github.com/bitcoin/bitcoin/pull/18115#pullrequestreview-362040407